src/Unleash/Scheduling/FetchFeatureTogglesTask.cs

```diff
@@ -74,17 +71,17 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
             if (result.Etag == Etag)
                 return;
 
-            toggleCollection.Instance = result.ToggleCollection;
+            if (!string.IsNullOrEmpty(result.State))
+            {
+                engine.TakeState(result.State);
```

Comment:
@daveleek This raises a new exception type, any idea where Exceptions that leave this method go? If they land in caller space then I'd like to wrap and reraise. If they get swallowed by async mechanics and become an error event then I don't care that much

Reply:
The primary use case is fine, gets called from the scheduler and outputs an error message in a log output. But this `ExecuteAsync` also gets called from `UnleashClientFactory.CreateClient` if `synchronousInitialization == true`, where errors are going straight back to caller space

Reply:
Is that intended? What should our plan be here?

Reply:
> The primary use case is fine, gets called from the scheduler and outputs an error message in a log output. But this `ExecuteAsync` also gets called from `UnleashClientFactory.CreateClient` if `synchronousInitialization == true`, where errors are going straight back to caller space

If I'm reading that correctly then yes, lands in user space

@nunogois sounds like 

```try{ 
  engine.TakeState(result.State);
} catch(YggException ex){
  throw new UnleashException(ex)
}```

or something very close is the correct thing here. Effectively what I don't want to happen here is code in the wild that's only catching `UnleashException` to have an YgggrasilException blow past it and unwind someone's stack

Reply:
Thanks! How about something like this? https://github.com/Unleash/unleash-client-dotnet/pull/224/commits/29650ef55ba60863197fb2f587350be5180acb44 - Should be consistent with how other exceptions are handled in this file.

Reply:
Awesome! Thanks @nunogois 👌